test/streaming-client.spec.ts

```diff
@@ -0,0 +1,188 @@
+import { PassThrough } from 'stream';
+import { WebSocketClient, WebSocketConnection } from 'websocket';
+import { AudioConfig } from '../src/models/streaming/AudioConfig';
+import { BufferedDuplex } from '../src/models/streaming/BufferedDuplex';
+import { RevAiStreamingClient } from '../src/streaming-client';
+const fs = require('fs');
+const events = require('events');
+
+jest.useFakeTimers();
+
+let sut: RevAiStreamingClient;
+let mockClient: WebSocketClient;
+
+const audioConfig = new AudioConfig("audio/x-wav");
+const token = "testToken";
+
+describe('streaming-client', () => {
+    beforeEach(() => {
+        sut = new RevAiStreamingClient(token, audioConfig);
+        mockClient = new WebSocketClient();
+        sut.client = mockClient;
+    });
+
+    describe('start', () => {
+        it('Connects to api with parameters', () => {
+            const res = sut.start();
+
+            expect(mockClient.connect).toBeCalledWith(`wss://api.rev.ai/speechtotext/v1alpha/stream?` +
+                `access_token=${token}` +
+                `&content_type=${audioConfig.getContentTypeString()}`
```

Comment:
Another way to write this test that decouples dependency on any implementation of getContentTypeString:
```
// Setup
const fakeAudioContentString = 'fakeAudioContentString';
audioConfig.getContentTypeString = jest.fn();
audioConfig.getContentTypeString.mockReturnValue(fakeAudioContentString);
...
// Assert
expect(audioConfig.getContentTypeString).toHaveBeenCalled();
...
`&content_type=${fakeAudioContentString}`
...
```

Reply:
I think we should write our ts tests like this from now on.

Reply:
Did you decide on wether to implement my suggested refactor?